apiserver/common/crossmodel/crossmodel.go

```diff
@@ -332,12 +332,14 @@ func ExpandChange(
 		})
 	}
 
+	uc := relation.UnitCount()
 	result := params.RemoteRelationChangeEvent{
 		RelationToken:       relationToken,
 		ApplicationToken:    appToken,
 		ApplicationSettings: appSettings,
 		ChangedUnits:        unitChanges,
 		DepartedUnits:       departed,
+		UnitCount:           &uc,
```

Comment:
Is the 2.8 facade version the same? Would this cause issues with a 2.7.7 model and a 2.8.0 controller? Or would perhaps it be fine because the cleanup is all controller side?

Reply:
This is a controller facade parameter - the api is only called by a worker in the same controller, so it has to be in sync by definition.
It is not called across controllers nor from agents to the controller, hence no need to up the facade version.
If libjuju is affected by this that's a bug we need to fix in libjuju.

Reply:
@howbazaar @wallyworld So both of you are right. It does impact libjuju, because any addition should be a facade bump, but @wallyworld is also right that this shouldn't be used by libjuju and should be treated as a bug.

In this circumstance we should be ok I believe to use `omitempty`.

I have another thought though. Juju itself has a package level separation of [Facades via packages](https://github.com/juju/juju/tree/develop/apiserver/facades) (ignoring the common Facade modules - sigh!), but this has never been lifted up into the code. The [`AllFacades`](https://github.com/juju/juju/blob/develop/apiserver/allfacades.go#L109) creates a `Registry`, which is a wrapper around `map[string]versions` (versions being the version of the facade and the factory method to create it`). I propose that we annotate the `Registry` either externally so you have many registries, or to the `Registry` itself. Adding an annotation about what type of Facade it is (`agent`, `controller`, or `client` and we could go more fine-grained here). 

From the Juju side you would still consume the `AllFacades` `Registry` in the same way,, so no difference there. The real difference is that for generating the schema for libjuju we would only generate all the `client` ones for example. This would cut down massively what we expose to the outside world. It would make libjuju smaller and lighter, the added benefit of it making it easier to understand the Juju API because of that factor. We don't expose APIs we don't want libjuju to interact with and it gives us more freedom with PRs like this.